catalog/content/core/content-core-videothumbnailplugin/src/test/java/ddf/content/plugin/video/TestVideoThumbnailPlugin.java

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.content.plugin.video;
+
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import javax.activation.MimeType;
+import javax.activation.MimeTypeParseException;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.SystemUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+
+import ddf.catalog.data.Metacard;
+import ddf.content.data.ContentItem;
+import ddf.content.operation.CreateRequest;
+import ddf.content.operation.CreateResponse;
+import ddf.content.operation.UpdateRequest;
+import ddf.content.operation.UpdateResponse;
+import ddf.content.plugin.PluginExecutionException;
+
+public class TestVideoThumbnailPlugin {
+    private String binaryPath;
+
+    private BundleContext mockBundleContext;
+
+    private VideoThumbnailPlugin videoThumbnailPlugin;
+
+    private ContentItem mockContentItem;
+
+    @Before
+    public void setUp() throws IOException, MimeTypeParseException, URISyntaxException {
+        System.setProperty("ddf.home", SystemUtils.USER_DIR);
+
+        binaryPath = FilenameUtils.concat(System.getProperty("ddf.home"), "bin_third_party");
+
+        setUpMockBundleContext();
+
+        videoThumbnailPlugin = new VideoThumbnailPlugin(mockBundleContext);
+
+        setUpMockContentItem();
+    }
+
+    private void setUpMockBundleContext() {
+        mockBundleContext = mock(BundleContext.class);
+
+        final Bundle mockBundle = mock(Bundle.class);
+        doReturn(mockBundle).when(mockBundleContext)
+                .getBundle();
+
+        String ffmpegResourcePath;
+        URL ffmpegBinaryUrl;
+
+        if (SystemUtils.IS_OS_LINUX) {
+            ffmpegResourcePath = "linux/ffmpeg";
+        } else if (SystemUtils.IS_OS_MAC) {
+            ffmpegResourcePath = "osx/ffmpeg";
+        } else if (SystemUtils.IS_OS_WINDOWS) {
```

Comment:
Solaris is one of the supported platforms for DDF. Can we include the Solaris ffmpeg binaries as well?